libraries/chain/include/eosio/chain/abi_def.hpp

```diff
@@ -1,8 +1,37 @@
 #pragma once
 
 #include <eosio/chain/types.hpp>
+#include <boost/algorithm/string/predicate.hpp>
 
-namespace eosio { namespace chain {
+namespace eosio::chain {
+
+struct version_t {
+   uint8_t major = 0;
+   uint8_t minor = 0;
+   bool    valid = false;
+
+   version_t(uint8_t major, uint8_t minor) : major(major), minor(minor), valid(true) {}
+
+   version_t(std::string_view sv) {
+      auto last = sv.data() + sv.size();
+      auto [ptr, ec] = std::from_chars(sv.data(), last, major);
+      if (ec == std::errc() && *ptr == '.')
+         valid = (std::from_chars(ptr+1, last, minor).ec == std::errc());
+   }
+
+   friend auto operator<=>(const version_t&, const version_t&) = default;
+   friend bool operator==(const version_t&, const version_t&) = default;
+
+   std::string str() const {
+      std::ostringstream oss;
+      oss << std::to_string(major) << '.' << std::to_string(minor);
+      return oss.str();
```

Comment:
`ostringstring` seems like a lot when this could just be `return std::to_string(major) + "." + std::to_string(minor);`

Reply:
https://github.com/AntelopeIO/spring/pull/1640/commits/4df6c8ceeab86c3126153a8740ab4d404bb73874